### PR TITLE
Adding CDM and DSBulk Migrator details

### PR DESCRIPTION
**Pull request by johnsmartco**

### CODE REVIEW

**Comment by johnsmartco** [comment]
Added overall data migrators info, and DSBulk Migrator commands/options details. Next up: CDM details - coming asap.

**Comment by johnsmartco** [comment]
These are still drafts but getting "closer to fine." Temp output is on coppi:

- http://coppi-ssh.aws.dsinternal.org/en/cdm4x-update/docs/migrate/migrate-and-validate-data.html  (parent page)
- http://coppi-ssh.aws.dsinternal.org/en/cdm4x-update/docs/migrate/cassandra-data-migrator.html
- http://coppi-ssh.aws.dsinternal.org/en/cdm4x-update/docs/migrate/dsbulk-migrator.html

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+== Install {cstar-data-migrator} as a Container
+
+Get the latest image that includes all dependencies from https://hub.docker.com/r/datastax/cassandra-data-migrator[DockerHub^].
+
+All migration tools (`cassandra-data-migrator` + `dsbulk` + `cqlsh`) are available in the `/assets/` folder of the container.
+
+[[cdm-install-as-jar]]
+== Install {cstar-data-migrator} as a JAR file
+
+Download the *latest* JAR file from the {cstar-data-migrator} https://github.com/datastax/cassandra-data-migrator/packages/[GitHub repo^].
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:31]:
```suggestion
Download the *latest* JAR file from the {cstar-data-migrator} https://github.com/datastax/cassandra-data-migrator/packages/[GitHub repo^]. image:https://img.shields.io/github/v/release/datastax/cassandra-data-migrator?color=green[Latest release]
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+
+[source,bash]
+----
+cd ~/github
+git clone git@github.com:datastax/cassandra-data-migrator.git
+cd cassandra-data-migrator
+mvn clean package
+----
+
+The fat jar (`cassandra-data-migrator-4.x.x.jar`) file should be present now in the `target` folder.
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:53]: Let's future-proof it 👍🏼 
```suggestion
The fat jar (`cassandra-data-migrator-x.y.z.jar`) file should be present now in the `target` folder.
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+2. Place the properties file that you elected to use and customize where it can be accessed while running the job via `spark-submit`.
+
+3. Run the job using `spark-submit` command:
+
+[source,bash]
+----
+./spark-submit --properties-file cdm.properties /
+--conf spark.cdm.schema.origin.keyspaceTable="<keyspacename>.<tablename>" /
+--master "local[*]" /
+--class com.datastax.cdm.job.Migrate cassandra-data-migrator-4.x.x.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:71]:
```suggestion
--class com.datastax.cdm.job.Migrate cassandra-data-migrator-x.y.z.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+====
+* The `spark-submit` command generates a log file, `logfile_name_*.txt`, to avoid log output on the Terminal console.
+* If the table you're migrating is large (such as over 100GB), you can add the option `--driver-memory 25G --executor-memory 25G`. Example:
+
+[source,bash]
+----
+./spark-submit --properties-file cdm.properties /
+--conf spark.cdm.schema.origin.keyspaceTable="<keyspacename>.<tablename>" /
+--master "local[*]" --driver-memory 25G --executor-memory 25G /
+--class com.datastax.cdm.job.Migrate cassandra-data-migrator-4.x.x.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:84]:
```suggestion
--class com.datastax.cdm.job.Migrate cassandra-data-migrator-x.y.z.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+
+To run your migration job with {cstar-data-migrator} in **data validation mode**, use class option `--class com.datastax.cdm.job.DiffData`. 
+Example:
+
+[source,bash]
+----
+./spark-submit --properties-file cdm.properties /
+--conf spark.cdm.schema.origin.keyspaceTable="<keyspacename>.<tablename>" /
+--master "local[*]" /
+--class com.datastax.cdm.job.DiffData cassandra-data-migrator-4.x.x.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:99]:
```suggestion
--class com.datastax.cdm.job.DiffData cassandra-data-migrator-x.y.z.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+All migration tools (`cassandra-data-migrator` + `dsbulk` + `cqlsh`) are available in the `/assets/` folder of the container.
+
+[[cdm-install-as-jar]]
+== Install {cstar-data-migrator} as a JAR file
+
+Download the *latest* JAR file from the {cstar-data-migrator} https://github.com/datastax/cassandra-data-migrator/packages/[GitHub repo^]. 
+
+[NOTE]
+====
+Version 4.x of {cstar-data-migrator} is not backward-compatible with *.properties files created in previous versions, and package names have changed. As of 21-Jun-2023, 4.1.0 is the latest version.
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:35]:
```suggestion
Version 4.x of {cstar-data-migrator} is not backward-compatible with `*.properties` files created in previous versions, and package names have changed. As of 21-Jun-2023, `4.1.0` is the latest version. If you're starting new, it is recommended to use the latest release version.
```

File: docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc [hunk 7]
```diff
@@ -0,0 +1,641 @@
...
+Alternatively, you can also pass the partition-file via a command-line parameter. 
+Example:
+
+[source,bash]
+----
+./spark-submit --properties-file cdm.properties /
+ --conf spark.cdm.schema.origin.keyspaceTable="<keyspacename>.<tablename>" /
+ --conf spark.cdm.tokenRange.partitionFile="/<path-to-file>/<csv-input-filename>" /
+ --master "local[*]" /
+ --class com.datastax.cdm.job.<Migrate|DiffData> cassandra-data-migrator-4.x.x.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```
Comment (msmygit) [inline docs-src/zdm-core/modules/migrate/pages/cassandra-data-migrator.adoc:161]:
```suggestion
 --class com.datastax.cdm.job.<Migrate|DiffData> cassandra-data-migrator-x.y.z.jar &> logfile_name_$(date +%Y%m%d_%H_%M).txt
```

**Review by msmygit — APPROVED** [review]
I've left some nitpick suggestions, but @johnsmartco has created a pure magic out of this project! 🙇🏼‍♂️ 🙏🏼 🚀

**Review by mendonk — APPROVED** [review]
+1

**Review by pravinbhat — APPROVED** [review]

**Pull request merged**